pkg/apiserver/profiling/router.go

```diff
@@ -260,13 +273,92 @@ func (s *Service) downloadSingle(c *gin.Context) {
 		return
 	}
 
-	fileName := fmt.Sprintf("profiling_%d.zip", taskID)
+	fileName := fmt.Sprintf("profiling_%s.zip", time.Now().Format("2006-01-02_15-04-05"))
 	c.Writer.Header().Set("Content-type", "application/octet-stream")
 	c.Writer.Header().Set("Content-Disposition", fmt.Sprintf("attachment; filename=\"%s\"", fileName))
-	err = ziputil.WriteZipFromFiles(c.Writer, []string{task.FilePath}, true)
+	zw := zip.NewWriter(c.Writer)
+	defer func() {
+		_ = zw.Close()
+	}()
+
+	err = writeZipFromFiles(zw, []string{task.FilePath}, true)
+	if err != nil {
+		_ = c.Error(err)
+		return
+	}
+
+	err = zipREADME(zw)
+	if err != nil {
+		_ = c.Error(err)
+		return
+	}
+}
+
+func writeZipFromFiles(zw *zip.Writer, files []string, compress bool) error {
+	for _, file := range files {
+		err := writeZipFromFile(zw, file, compress)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func writeZipFromFile(zw *zip.Writer, file string, compress bool) error {
+	f, err := os.Open(filepath.Clean(file))
+	if err != nil {
+		return err
+	}
+	defer func() {
+		_ = f.Close()
+	}()
+
+	fileInfo, err := f.Stat()
+	if err != nil {
+		return err
+	}
+
+	zipMethod := zip.Store // no compress
+	if compress {
+		zipMethod = zip.Deflate // compress
+	}
+	zipFile, err := zw.CreateHeader(&zip.FileHeader{
+		Name:     fileInfo.Name(),
+		Method:   zipMethod,
+		Modified: time.Now(),
+	})
+	if err != nil {
+		return err
+	}
+
+	_, err = io.Copy(zipFile, f)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func zipREADME(zw *zip.Writer) error {
+	const downloadREADME = `
+To review the CPU profiling or heap profiling result interactively:
+
+$ go tool pprof --http=0.0.0.0:1234 cpu_xxx.proto
+`
+	zipFile, err := zw.CreateHeader(&zip.FileHeader{
+		Name:     "README.md",
+		Method:   zip.Deflate,
+		Modified: time.Now(),
+	})
+	if err != nil {
+		return err
+	}
+
+	_, err = zipFile.Write([]byte(downloadREADME))
 	if err != nil {
-		log.Error("Stream zip pack failed", zap.Error(err))
+		return err
 	}
+	return nil
 }
```

Comment:
How about passing the content and file name from outside, so that become a utility method. We can continue to refine it later.

Reply:
Yes, we can refine this later.